docs/src/tutorials/Systems and Control/barrier_certificate.jl

```diff
@@ -0,0 +1,77 @@
+# # Barrier certificates for collision avoidance
+
+# **Adapted from**: Example 2 of "Engineering applications of SOS polynomials" by Georgina Hall, from 
+# the 2019 AMS Short Course on Sum of Squares: Theory and Applications
+
+using Test #src
+using DynamicPolynomials
+@polyvar x[1:2]
+
+using SumOfSquares
+using CSDP
+
+# We need to pick an SDP solver, see [here](https://jump.dev/JuMP.jl/v1.8/installation/#Supported-solvers) for a list of the available choices.
+# We use `SOSModel` instead of `Model` to be able to use the `>=` syntax for Sum-of-Squares constraints.
+solver = optimizer_with_attributes(CSDP.Optimizer, MOI.Silent() => true)
+model = SOSModel(solver);
+
+# We define below the vector field ``\text{d}x/\text{d}t = f``
+
+f = [ x[2],
+     -x[1] + (1/3)*x[1]^3 - x[2]]
+
+# We now define the functions that describes the initial and unsafe sets
+
+g₁ = -(x[1]+1)^2 - (x[2]+1)^2 + 0.16
+h₁ = -(x[1]-1.5)^2 - x[2]^2 + 0.25
+
+# Define SOS barrier function B
+monos = monomials(x, 0:4)
+@variable(model, B, Poly(monos))
+
+# Define barrier certificate constraints 
+
+# B(x) = ε + σ₀(x) + σ₁(x)g₁(x) where σᵢ are SOS and ε > 0
+ε = 0.001
+@variable(model, σ₁, Poly(monos))
```

Comment:
Here it's not created as SOS, I recommend using the `@set` macro and `domain` keyword argument to the macro below

Reply:
Ah, nice! I had previously believed that their usage was limited to polynomial optimization scenarios.